pkg/client/interceptor/interceptor.go

```diff
@@ -4,3 +4,191 @@
 
 // Package interceptor contains gRPC client interceptors related to authentication and authorization.
 package interceptor
+
+import (
```

Comment:
This appears as a new file, but actually, it is a move & refactor of the `signature.go`.

I renamed the types and their methods for better naming.